homeassistant/components/sensor/zoneminder.py

```diff
@@ -121,22 +113,9 @@ def state(self):
 
     def update(self):
         """Update the sensor."""
-        date_filter = '1%20{}'.format(self._type)
-        if self._type == 'all':
-            # The consoleEvents API uses DATE_SUB, so give it
-            # something large
-            date_filter = '100%20year'
-
-        archived_filter = '/Archived=:0'
-        if self._include_archived:
-            archived_filter = ''
-
-        event = zoneminder.get_state(
-            'api/events/consoleEvents/{}{}.json'.format(date_filter,
-                                                        archived_filter)
-        )
-
-        try:
-            self._state = event['results'][str(self._monitor_id)]
-        except (TypeError, KeyError):
-            self._state = '0'
+        events = self._monitor.get_events(
+            self.time_period, self._include_archived)
+        if not events:
+            self._state = 0
```

Comment:
What does 0 mean for this state? Isn't it better to use `None` which will show as unknown state?

Reply:
Yeah you are correct, the library will return `None` or a number, so directly setting that to `self._state` should be sufficient